src/snowflake/cli/_app/telemetry.py

```diff
@@ -81,6 +86,37 @@ class TelemetryEvent(Enum):
 TelemetryDict = Dict[Union[CLITelemetryField, TelemetryField], Any]
 
 
+def _is_cli_exception(exception: Exception) -> bool:
+    return isinstance(
+        exception,
+        (click.ClickException, typer.Exit, typer.Abort, BrokenPipeError),
+    )
+
+
+def _get_additional_exception_information(exception: Exception) -> TelemetryDict:
+    """
+    Attach the errno and sqlstate if the exception or the
+    cause of the exception is a ProgrammingError
+    """
+    additional_info = {}
+
+    if isinstance(exception, ProgrammingError):
+        additional_info[CLITelemetryField.ERROR_CODE] = exception.errno
```

Comment:
Can we please cast it to integer?

Reply:
The type hints say it already should be an integer or None, in which case the casting would throw an error:
<img width="317" alt="image" src="https://github.com/user-attachments/assets/f7fa1380-b4eb-4998-a2b9-7d410f012b7d">